gitpod-io/gitpod
# [server] listWorkspaceSessions API

## Description

Introduces a new RPC for listing workspaces sessions (aka as workspace instances internally).
The API uses the same workspace shape which is already a shape that combines information from a workspace and a workspaceInstance (usually the latest one).

## Related Issue(s)

Fixes https://linear.app/gitpod/issue/ENT-66/add-listworkspacesessions-rpc-call-to-workspace-service-endpoint

## How to test


https://se-workspad026375d1b.preview.gitpod-dev.com/workspaces

Create a bunch of workspaces and also restart them in order to create multiple instances per workspace.
Ideally also do this with different users in the same work.

Use the new API to list the sessions and verify the data represents the system state.

```
curl -X POST \
  'https://api.gitpod.io/gitpod.v1.WorkspaceService/ListWorkspaceSessions' \
  --header 'Authorization: Bearer <YOUR_ACCESS_TOKEN>' \
  --header 'content-type: application/json' \
  --data-raw '{
      "organizationId": "<ORG_ID>"
}'
  ```

## Documentation


#### Preview status

gitpod:summary

## Build Options

<details>
<summary>Build</summary>

      Run the build with werft instead of GHA
      Run Leeway with `--dont-test`
</details>

<details>
<summary>Publish</summary>

</details>

<details>
<summary>Installer</summary>

  Add desired feature flags to the end of the line above, space separated
</details>

<details>
<summary>Preview Environment / Integration Tests</summary>

      If enabled this will build `install/preview`
      If enabled this will create the environment on GCE infra
      Saves cost. Untick this only if you're really sure you need a non-preemtible machine.
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`. If enabled, `with-preview` and `with-large-vm` will be enabled.
</details>

/hold
